### PR TITLE
Dyn 4583 group style name mandatory

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

When adding a new GroupStyle if the group name TextBox was not populated then was auto inserting a text, now with this fix the group name TextBox will be mandatory if is not populated and the Save button is pressed when we will show a warning message. the user will able to create the style when the group name is populated and has a unique value.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Make the groupstyle name TextBox a mandatory field.

### Reviewers

@QilongTang

### CODE REVIEW

**Comment by reddyashish** [comment]
@RobertGlobant20 Can you add a gif or screenshot to describe the changes?

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs [hunk 1]
```diff
@@ -177,18 +185,26 @@ private void AddStyle_SaveButton_Click(object sender, RoutedEventArgs e)
...
-                viewModel.AddStyle(newItem);
-                viewModel.ResetAddStyleControl();
-                AddStyleBorder.Visibility = Visibility.Collapsed;
-                AddStyleButton.IsEnabled = true;
+                viewModel.CurrentWarningMessage = Res.PreferencesViewEmptyStyleWarning;
+                viewModel.IsWarningEnabled = true;
+                viewModel.IsSaveButtonEnabled = false;
             }
-            else
+            else if (viewModel.ValidateExistingStyle(newItem) == true)
```
Comment (reddyashish) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:194]: The If condition here can be changed to if(viewModel.ValidateExistingStyle(newItem))

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs [hunk 2]
```diff
@@ -177,18 +185,26 @@ private void AddStyle_SaveButton_Click(object sender, RoutedEventArgs e)
...
+                viewModel.IsSaveButtonEnabled = false;
             }
-            
+            else
+            {
+                viewModel.AddStyle(newItem);
+                viewModel.ResetAddStyleControl();
+                AddStyleBorder.Visibility = Visibility.Collapsed;
+                AddStyleButton.IsEnabled = true;
+                viewModel.IsSaveButtonEnabled = true;
```
Comment (reddyashish) [inline src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:206]: This is already done inside ResetAddStyleControl(), right?

**Comment by RobertGlobant20** [comment]
> @RobertGlobant20 Can you add a gif or screenshot to describe the changes?

Here is a GIF of the functionality.
group_name_mandatory

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs (hunk 2 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:206]: @reddyashish ResetAddStyleControl() method is doing other stuff.
image

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs (hunk 1 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs]: @reddyashish updated in the next commit:
https://github.com/DynamoDS/Dynamo/pull/12691/commits/59650714612e797d5bb8097f631f606a80ad15ea

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs (hunk 2 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:206]: @RobertGlobant20 Then should we add these into the reset function as well?

Comment (reddyashish) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs:206]: Oh yeah I got confused with the ResetGroupStyleForm().

**Comment by reddyashish** [comment]
@RobertGlobant20 Looks good to me. The only suggestion I have is, it would be great if we can remove the warning(please enter a valid name) and the icon, when the user enters the first character. We can check for the 'name already in use' case when the save button is clicked. It would be more clear that way to the user. If you think it would need more changes, I think we can ignore it. @QilongTang What do you think?

**Comment by QilongTang** [comment]
> @RobertGlobant20 Looks good to me. The only suggestion I have is, it would be great if we can remove the warning(please enter a valid name) and the icon, when the user enters the first character. We can check for the 'name already in use' case when the save button is clicked. It would be more clear that way to the user. If you think it would need more effort, we can ignore it. @QilongTang What do you think?

@reddyashish Do you suggest decoupling the warning state with the validation? After the warning icon appear, clearning the text would remove the icon?

**Comment by reddyashish** [comment]
Yes, we can remove the warning icon either after clearing the default name text or after the user enters the first character in the name field.

File: src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs (hunk 2 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs]: @QilongTang I added those lines in the ResetGroupStyleControl() method and also I refactored some code, please check the next commit: https://github.com/DynamoDS/Dynamo/pull/12691/commits/66758055b162acac6f5bdf7d110062eb2c55479a

**Comment by RobertGlobant20** [comment]
> Yes, we can remove the warning icon either after clearing the default name text or after the user enters the first character in the name field.

@reddyashish the functionality was updated, so now if the user type something in the textbox when the is a warning message, it will disappear and the validation will be executed again when the Save button is clicked.
see the next commit: https://github.com/DynamoDS/Dynamo/pull/12691/commits/66758055b162acac6f5bdf7d110062eb2c55479a

**Comment by RobertGlobant20** [comment]
GIF with the new functionality updated.
group_name_mandatory_codereview1

**Comment by QilongTang** [comment]
Thank you @RobertGlobant20 @reddyashish , I can see the improvement there for user experience, will take another look at this PR

**Review by QilongTang — COMMENTED** [review]
@reddyashish @zeusongit This looks fine to me, what do you think?

**Review by reddyashish — APPROVED** [review]
Looks good to me. Thanks @RobertGlobant20

**Pull request merged**